### PR TITLE
chore(*): Cleans up unneeded allocations in our providers

### PR DESCRIPTION
**Pull request by thomastaylor312**
This should allow them to run even slimmer in constrained environments.
Please note I did not bump versions on the providers as there is an
incoming wasmbus-rpc change and there isn't any rush to get these out
right now

### CODE REVIEW

**Comment by stevelr** [comment]
at first glance these all look pretty good. Did you run all the tests ('make test' from top-level dir)? blobstore looks like it had the most changes that might affect tests. I agree they shouldn't be version bumped until we roll in wasmbus-rpc 0.10.

File: blobstore-s3/src/lib.rs [hunk 1]
```diff
@@ -644,6 +638,7 @@ impl Blobstore for StorageClient {
                 "cannot put zero-length objects".to_string(),
             ));
         }
+        // TODO: make sure put_object takes an owned `PutObjectRequest` to avoid cloning the whole chunk
```
Comment (brooksmtownsend) [inline blobstore-s3/src/lib.rs:641]: Is this something that was intended to make it into this PR?

File: httpserver-rs/src/settings.rs [hunk 2]
```diff
@@ -4,6 +4,7 @@
 //! which determine how the configuration is parsed.
 //!
 //! For the key...
+use std::net::{IpAddr, Ipv4Addr};
```
Comment (brooksmtownsend) [inline httpserver-rs/src/settings.rs:7]: should this be with the other imports?

**Review by brooksmtownsend — APPROVED** [review]
All looks reasonable 😄

**Comment by brooksmtownsend** [comment]
> at first glance these all look pretty good. Did you run all the tests ('make test' from top-level dir)? blobstore looks like it had the most changes that might affect tests. I agree they shouldn't be version bumped until we roll in wasmbus-rpc 0.10.

@stevelr `make test` is run as a part of the [blobstore s3 action](https://github.com/wasmCloud/capability-providers/runs/7531347044?check_suite_focus=true) now, so it looks like it all passed with flying colors

File: blobstore-s3/src/lib.rs (hunk 1 above)
Comment (thomastaylor312) [reply blobstore-s3/src/lib.rs:641]: Yes, it was a note that we need to fix it

File: httpserver-rs/src/settings.rs (hunk 2 above)
Comment (thomastaylor312) [reply httpserver-rs/src/settings.rs:7]: Std imports are separated by a line before external crates import, followed by another line before internal imports

**Pull request merged**